src/Strategy/Python/Pipenv.hs

```diff
@@ -71,7 +74,7 @@ strategyWithCmd = Strategy
 strategyNoCmd :: Strategy BasicFileOpts
 strategyNoCmd = Strategy
   { strategyName = "python-pipfile"
-  , strategyAnalyze = \opts -> analyzeNoCmd & fileInputJson (targetFile opts)
+  , strategyAnalyze = \opts -> analyzeNoCmd & (fileInputJson :: (Members '[ReadFS, Error CLIErr] r) => Path b File -> InterpreterFor (Input PipfileLock) r) (targetFile opts)
```

Comment:
There's a really useful language extension called `TypeApplications` that lets you specify concrete types for type variables:

For example, with `const`:
```hs
> :type const
a -> b -> a
> :type const @Int
Int -> b -> Int
> :type const @Int @String
Int -> String -> Int
```

We use this [elsewhere in the codebase](https://github.com/fossas/spectrometer/blob/master/src/App/Scan.hs#L55) when disambiguating types

You should be able to use something similar here with `fileInputJson @... (targetFile opts)`